is_core/auth_token/utils.py

```diff
@@ -78,14 +101,29 @@ def dont_enforce_csrf_checks(request):
                 request.META.get(header_name_to_django(config.IS_CORE_AUTH_HEADER_NAME)))
 
 
+def get_user_from_token(token):
+    if token:
+        backend_path = token.backend
+        if backend_path in settings.AUTHENTICATION_BACKENDS:
+            active_takeover_id = token.active_takeover.user.pk if token.active_takeover else None
+            user_id = token.user.pk
+            backend = load_backend(backend_path)
+            return backend.get_user(active_takeover_id) or backend.get_user(user_id) or AnonymousUser()
+    return AnonymousUser()
+
+
 def get_user(request):
     """
     Returns the user model instance associated with the given request token.
     If no user is retrieved an instance of `AnonymousUser` is returned.
     """
-    if hasattr(request, 'token'):
-        user_id = request.token.user.pk
-        user = ModelBackend().get_user(user_id) or AnonymousUser()
+    return get_user_from_token(getattr(request, 'token'))
+
+
+def takeover(request, user):
+    if request.user == user:
+        return False
     else:
-        user = AnonymousUser()
-    return user
+        request.token.user_takeovers.update(is_active=False)
+        request.token.user_takeovers.create(user=user, is_active=True)
+        return True
```

Comment:
Tak nic, ty všechny takeovers vypneš, když vytváříš nový, tím pádem to všechno funguje dobře.